File: content/atlas/source/includes/fts/partial-match/QueryWildcard.cs
using MongoDB.Bson;
using MongoDB.Driver;
using System;
using System.Threading.Tasks;

namespace MongoSearchExamples
{
    class QueryWildcard
    {
        // Connection URI for your Atlas deployment
        private static readonly string connectionUri = "<connection-string>";

        static async Task Main(string[] args)
        {
            var client = new MongoClient(connectionUri);
            
            try
            {
                // Set namespace
                var database = client.GetDatabase("sample_mflix");
                var collection = database.GetCollection<BsonDocument>("movies");

                // Define pipeline
                var pipeline = new BsonDocument[]
                {
                    new BsonDocument("$search", new BsonDocument
                    {
                        { "index", "partial-match-tutorial" },
                        { 
                            "wildcard", new BsonDocument
                            {
                                { "path", "title" },
                                { "query", "how*" },
                                { "allowAnalyzedField", true }
                            }
                        }
                    }),
                    new BsonDocument("$project", new BsonDocument
                    {
                        { "_id", 0 },
                        { "title", 1 }
                    })
                };

                // Run pipeline
                var result = collection.Aggregate<BsonDocument>(pipeline);
                
                // Print results
                await result.ForEachAsync(doc => Console.WriteLine(doc.ToJson()));
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Error: {ex.Message}");
            }
        }
    }
}

File: content/atlas/source/includes/fts/partial-match/queryWildcard.py
from pymongo import MongoClient

# connect to your Atlas cluster
uri = "<connection-string>"

client = MongoClient(uri)

def run():
    try:
        # set namespace
        database = client["sample_mflix"]
        coll = database["movies"]

        # define pipeline
        agg = [
            {
                "$search": {
                    "index": "partial-match-tutorial",
                    "wildcard": {
                        "path": "title",
                        "query": "how*",
                        "allowAnalyzedField": True,
                    }
                }
            },
            {
                "$project": {
                    "_id": 0,
                    "title": 1
                }
            }
        ]

        # run pipeline
        result = coll.aggregate(agg)
        print(result)

        # print results
        for doc in result:
            print(doc)
    finally:
        client.close()

if __name__ == "__main__":
    run()
